src/pkcs15init/pkcs15-myeid.c

```diff
@@ -908,12 +917,14 @@ myeid_generate_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
 			pubkey->u.ec.params.named_curve = strdup(ecparams->named_curve);
 			if (!pubkey->u.ec.params.named_curve)
 				LOG_FUNC_RETURN(ctx, SC_ERROR_OUT_OF_MEMORY);
+
 			r = sc_pkcs15_fix_ec_parameters(ctx, &pubkey->u.ec.params);
 			LOG_TEST_RET(ctx, r, "Cannot fix EC parameters");
 		}
 	}
 
-	sc_file_free(file);
+	if (file)
+		sc_file_free(file);
```

Comment:
The `sc_file_free()` should correctly handle NULL arguments so there is no need for this change.